### PR TITLE
Keep SNMP values also in memory to avoid file reading

### PR DESCRIPTION
**Pull request by pyhys**
Implement proper functions for other parts of the stack to properly
interact with SNMP.

Found with ART Tester case "Topology discovery check (non-PN)" and a
device running on Raspberry Pi (with an almost full SD card file system)

Use full path for the cmake generator expression in src/CMakeLists.txt and
test/CMakeLists.txt to work with cmake 3.19.2

### CODE REVIEW

**Review by olbjo — APPROVED** [review]
👍 
Looks great. A minor comment that you can ignore if you disagree

File: src/common/pf_snmp.h [hunk 1]
```diff
@@ -251,6 +253,13 @@ typedef struct pf_snmp_signal_delay
    uint32_t line_propagation_delay_ns;
 } pf_snmp_signal_delay_t;
 
+/**
+ * Initialize SNMP related data.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/common/pf_snmp.h]: Updated comment.
Done.

File: src/common/pf_snmp.c [hunk 2]
```diff
@@ -225,25 +236,123 @@ static void pf_snmp_encode_signal_delays (
...
-   CC_STATIC_ASSERT (sizeof (name->string) >= PNAL_HOSTNAME_MAX_SIZE);
+   CC_STATIC_ASSERT (
+      sizeof (net->snmp_data.system_name.string) >= PNAL_HOSTNAME_MAX_SIZE);
 
-   error = pf_file_load (directory, PF_FILENAME_SYSNAME, name, sizeof (*name));
+   /* sysContact */
+   error = pf_file_load (
+      directory,
+      PF_FILENAME_SNMP_SYSCONTACT,
+      &net->snmp_data.system_contact,
```
Comment (hefloryd) [inline src/common/pf_snmp.c:252]: Suggestion: use a temporary variable to reduce repetition, e.g snmp_data = &net->snmp_data. This makes the code somewhat easier to follow.

Comment (pyhys) [reply src/common/pf_snmp.c]: Introduced
`pf_snmp_data_t * snmp = &net->snmp_data;`
for` pf_snmp_data_init()` and `pf_snmp_data_clear()` 
Done

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -14,6 +14,8 @@
 #*******************************************************************/
 
 # NOTE: add headers to make them show up in an IDE
+# NOTE: Use full path for the <$<BOOL:${PNET_OPTION_SNMP}> expression
+#       to work with cmake 3.19.2
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/CMakeLists.txt]: Changed to 
```
# NOTE: Use full path for the <$<BOOL:${PNET_OPTION_SNMP}> expression
#       to work with certain cmake versions
```
Done

**Pull request merged**